controller/gatewayclass/controller.go

```diff
@@ -55,29 +53,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
-	if !gwc.IsAccepted() {
-		oldGwc := gwc.DeepCopy()
+	oldGwc := gwc.DeepCopy()
 
-		k8sutils.SetCondition(
-			k8sutils.NewConditionWithGeneration(
-				consts.ConditionType(gatewayv1.GatewayClassConditionStatusAccepted),
-				metav1.ConditionTrue,
-				consts.ConditionReason(gatewayv1.GatewayClassReasonAccepted),
-				"the gatewayclass has been accepted by the operator",
-				gwc.GetGeneration(),
-			),
-			gwc,
-		)
-		if err := r.Status().Patch(ctx, gwc.GatewayClass, client.MergeFrom(oldGwc)); err != nil {
-			if k8serrors.IsConflict(err) {
-				log.Debug(logger, "conflict found when updating GatewayClass, retrying")
-				return ctrl.Result{
-					Requeue:      true,
-					RequeueAfter: controller.RequeueWithoutBackoff,
-				}, nil
-			}
-			return ctrl.Result{}, fmt.Errorf("failed patching GatewayClass: %w", err)
+	condition, err := getAcceptedCondition(ctx, r.Client, gwc.GatewayClass)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get accepted condition: %w", err)
```

Comment:
Shouldn't we set the condition here with some error message so that users do not have to look into operators logs? Ideally I'd consider adding the error's message into the condition message so that users know what happened. WDYT?

Reply:
The `AcceptedCondition` message is properly set in the `getAcceptedCondition` function. Isn't this what you are referring to?

Reply:
I was specifically alluding here to lines 42-44 https://github.com/Kong/gateway-operator/pull/1021/files/31ff623213f0fed16947060e8a0eccec5e02f4ff#diff-f4e83572e27854d40d56ef54e8f982cdca3ccaaa3f17900c3ab6e3f4fb04e971R42-R44 where the `err` is simply returned. Then we seem to return it here as well. Not setting anything on the `GatewayClass`. Or am I missing something here?

Reply:
That can be only a transient error caused by the get function. I don't think we should change the status of the resource when such an error is raised. Do you?

Reply:
It doesn't have to be transient. It can also be a problem with e.g. permissions. I believe it would be better UX to make the error visible in the status rather than forcing users to go through operator's logs. WDYT?

Reply:
I don't think so honestly. If that's an error related to permissions, for example, that isn't related to the resource, but to the operator itself. An object status should include information related to the resource(s) themselves, not to the operators managing it IMO.

Reply:
> If that's an error related to permissions, for example, that isn't related to the resource, but to the operator itself. 

I agree but the error in object's status is a breadcrumb which can help you as a user debug this problem faster.

The permission might not be directly related to the operator. There might exist policies in the cluster which would prevent the `Get` from succeeding.

I have already seen reports of several issues where users were confused because an object didn't get any status condition set (for different reasons but still). For this reason I do believe that putting information that can help debug a problem in status condition is going to allow people to quickly resolve an issue that caused it.

Reply:
> But here we are discussing about putting information that has nothing to do with the resource itself.

Agreed. Nevertheless this information is beneficial to the user to quickly resolve an issue with this object not being accepted.

> I think we should instead try to educate users about what kind of information can be found where.

Agreed on this as well. Do you have an idea how we could educate users about this one so that they can better debug problems at we discuss here?

---

I see your view and I understand it yet I do not share it. Feel free to continue with the way this is proposed as of now 👍